packages/tspec/src/generator/index.ts

```diff
@@ -157,9 +157,15 @@ const getOpenapiSchemasOnly = (openapiSchemas: SchemaMapping, tspecSymbols: stri
   });
 
   return Object.fromEntries(
-    Object.entries(openapiSchemas).filter(
-      ([key]) => (!tspecSymbols.includes(key) && !tspecPathSchemas.includes(key)),
-    ),
+    Object.entries(openapiSchemas)
+      .filter(
+        ([key]) =>
+          !tspecSymbols.includes(key) && !tspecPathSchemas.includes(key)
+      )
+      .map(([key, value]: [key: string, value: OpenAPIV3.SchemaObject & {mediaType?: string} ]) => {
+        const { mediaType = "", ...body } = value || {};
+        return [key, body];
+      })
```

Comment:
This logic became too complex, so I refactored it by breaking it into functions. https://github.com/ts-spec/tspec/pull/35/commits/1cc6061aa1e61d7bb8da7d9e0fa4f2080226fa5b